utam-core/src/main/java/utam/core/framework/base/ElementBuilder.java

```diff
@@ -40,15 +41,29 @@
    * @param <T>    element type
    * @return instance with parameters set in selector
    */
-  public <T extends Actionable> T build(Class<T> type, Object... values) {
+  public <T extends Actionable, R extends BasePageElement> T build(
+      Class<T> type, Class<R> implType, Object... values) {
     ElementLocation elementLocation = this.elementFinder.setParameters(values);
     Element element = factory.findElement(elementLocation);
     if(element.isNull()) {
       return null;
     }
-    return (T) new BasePageElement(factory, element);
+    return createInstance(implType, element);
   }
 
+  private <T extends Actionable, R extends BasePageElement> T createInstance(
+      Class<R> implType, Element element) {
+    R result = null;
+    try {
+      result = implType.getConstructor().newInstance();
+      result.initialize(factory, element);
+    } catch (ReflectiveOperationException e) {
+      return null;
```

Comment:
why we return null? i'd throw error here

Reply:
My thinking was to help preserve the contract of `build()` which returns `null`. I suppose throwing is a fine approach also. I'll update to do that.

Reply:
ah, good point. 
we can return null if condition element.isNull() is met
Reflection exception on the other hand means that something went wrong so i think we should throw it